keras/src/backend/jax/nn.py

```diff
@@ -659,3 +659,15 @@ def _iterate(prev, x):
     )
 
     return -last_alpha_mask[jnp.arange(batch_size), target_length]
+
+
+def psnr(x, y, max_val):
+    if x.shape != y.shape:
+        raise ValueError(
+            f"Expected x and y to have the same shape. Got x: {x.shape} and y: {y.shape}."
+        )
+
+    max_val = convert_to_tensor(max_val, x.dtype)
+    mse = jnp.mean(jnp.square(x - y), axis=(-3, -2, -1))
```

Comment:
We should just remove the axis argument -- right now it presupposes that the input has at least 3 axes, but in practice the input could have any rank >= 2.